### PR TITLE
More utilities

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: downloader/src/main/kotlin/vcs/Git.kt [hunk 1]
```diff
@@ -121,7 +122,7 @@ class Git : GitBase() {
         }
 
         // Fall back to fetching everything.
-        return try {
+        return succeeds({
```
Comment (mnonnenmacher) [inline downloader/src/main/kotlin/vcs/Git.kt]: Using the `succeeds` help function instead of catching a specific exception can hide bugs, because the generic `Throwable` is catched in this case. Also IMO the variant with two lambda arguments makes it harder to read the code, because it's unintuitive. I would suggest to make the helper function internal and to only use it in the simple cases of the extension functions in `Extensions.kt`.

File: spdx-utils/src/main/kotlin/SpdxUtils.kt [hunk 2]
```diff
@@ -50,16 +48,21 @@ val NON_LICENSE_FILENAMES = listOf(
     "META-INF/DEPENDENCIES"
 )
 
+/**
+ * Return a string of hexadecimal digits representing the bytes in the array.
+ */
+private fun ByteArray.toHexString(): String = joinToString("") { String.format("%02x", it) }
```
Comment (mnonnenmacher) [inline spdx-utils/src/main/kotlin/SpdxUtils.kt]: The title and body of the commit message talk of two different changes, suggesting it should have been two commits instead. Also, the change in `SpdxUtilsTest` seems unrelated.

File: utils/src/main/kotlin/Extensions.kt [hunk 3]
```diff
@@ -41,10 +41,9 @@ import java.nio.file.Path
 import java.nio.file.SimpleFileVisitor
 import java.nio.file.StandardCopyOption
 import java.nio.file.attribute.BasicFileAttributes
+import java.security.MessageDigest
```
Comment (mnonnenmacher) [inline utils/src/main/kotlin/Extensions.kt]: That change requires more explanation in the commit message, I guess it's something along the lines of "commons codec only provides a nicer interface for `MessageDigest`, but no additional functionality".

File: downloader/src/main/kotlin/vcs/Cvs.kt [hunk 4]
```diff
@@ -29,14 +29,13 @@ import com.here.ort.utils.ProcessCapture
 import com.here.ort.utils.log
 import com.here.ort.utils.safeDeleteRecursively
 import com.here.ort.utils.searchUpwardsForSubdirectory
+import com.here.ort.utils.toHexString
```
Comment (mnonnenmacher) [inline downloader/src/main/kotlin/vcs/Cvs.kt]: Same as previous commit.

File: spdx-utils/src/main/kotlin/SpdxDeclaredLicenseMapping.kt [hunk 5]
```diff
@@ -19,6 +19,7 @@
 
 package com.here.ort.spdx
 
+import com.here.ort.spdx.SpdxExpression.Strictness
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply spdx-utils/src/main/kotlin/SpdxDeclaredLicenseMapping.kt]: Fixed.

File: spdx-utils/src/main/kotlin/SpdxDeclaredLicenseMapping.kt [hunk 6]
```diff
@@ -312,6 +313,23 @@ object SpdxDeclaredLicenseMapping {
         "the Apache License, ASL Version 2.0" to APACHE_2_0.toExpression()
     )
 
+    init {
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply spdx-utils/src/main/kotlin/SpdxDeclaredLicenseMapping.kt]: Fixed as part of #1799.

File: downloader/src/main/kotlin/vcs/Git.kt (hunk 1 above)
Comment (sschuberth) [reply downloader/src/main/kotlin/vcs/Git.kt]: Solved almost like that: I left the function public in case someone still wants to use it form the outside.

File: spdx-utils/src/main/kotlin/SpdxUtils.kt (hunk 2 above)
Comment (sschuberth) [reply spdx-utils/src/main/kotlin/SpdxUtils.kt]: I've split out the `SpdxUtilsTest` change, but the `toHexString()` change *is* actually part of the topic as explained in the commit message:

> toHexString() is copied from the utils module as we do not want spdx-utils to depend on it, so that spdx-utils could be published independently.

`toHexString()` replaces `Hex.encodeHexString()` from `commons-codec`.

File: utils/src/main/kotlin/Extensions.kt (hunk 3 above)
Comment (sschuberth) [reply utils/src/main/kotlin/Extensions.kt]: I've added more explanatory text to the message of the commit before, as that's actually the first commit introducing this kind of change.
